flink-metrics/flink-metrics-prometheus/src/main/java/org/apache/flink/metrics/prometheus/PrometheusReporter.java

```diff
@@ -50,30 +49,11 @@ int getPort() {
         return port;
     }
 
-    @Override
-    public void open(MetricConfig config) {
-        super.open(config);
-
-        String portsConfig = config.getString(ARG_PORT, DEFAULT_PORT);
-        Iterator<Integer> ports = NetUtils.getPortRangeFromString(portsConfig);
-
-        while (ports.hasNext()) {
-            int port = ports.next();
-            try {
-                // internally accesses CollectorRegistry.defaultRegistry
-                httpServer = new HTTPServer(port);
-                this.port = port;
-                log.info("Started PrometheusReporter HTTP server on port {}.", port);
-                break;
-            } catch (IOException ioe) { // assume port conflict
-                log.debug("Could not start PrometheusReporter HTTP server on port {}.", port, ioe);
-            }
-        }
-        if (httpServer == null) {
-            throw new RuntimeException(
-                    "Could not start PrometheusReporter HTTP server on any configured port. Ports: "
-                            + portsConfig);
-        }
+    PrometheusReporter(
+        final int port, 
+        @Nullable HTTPServer httpServer) {
+        this.httpServer = Preconditions.checkNotNull(httpServer);
```

Comment:
`@Nullable` to us implies that null is an expected case that should be acceptable. Hence, it doesn't make to annotate it as such if we then right away enforce that it is in fact not null.

Reply:
Makes sense, although the `httpServer` is no longer passed in since we pass in the ports iterator now. Done!